_posts/en/newsletters/2019-11-27-newsletter.md

```diff
@@ -0,0 +1,301 @@
+    exchange nonce commitments and then nonces before they know what
+    transaction they want to commit to with their signature---but this
+    is unsafe due to Wagner's algorithm, as Nick briefly explains.  The
+    only information that can be safely shared before each participant
+    knows the transaction to sign is the nonce commitment.  (Not
+    mentioned in the blog post, but discussed on IRC, was that Pieter
+    Wuille and others have been investigating a construction based on a
+    Zero Knowledge Proof (ZKP) that could allow reduced interactivity.)
+    The blog post concludes with a suggestion that interested readers
+    review the MuSig implementation in [libsecp256k1-zkp][], which is
+    designed to help developers use the protocol safely.
+
+    Influenced by Jonas Nick's presentation on this topic at the Berlin
+    Lightning Conference, Adam Gibson wrote a separate [blog
+    post][gibson wagners] that describes Wagner's algorithm in much more
+    detail with a mix of math, intuitive analysis, and topical
+    information Bitcoiners may find interesting (such as the amusing
+    tidbit of [Wagner's paper][] citing both Adam Back and Wei Dai
+    several years before Nakamoto did [the same][bitcoin.pdf], though
+    for different work).  Anyone interested in the developing their own
+    cryptographic protocols is recommended to read both posts, as each
+    compliments the other without being repetitive about the subject.
+
+## Selected Q&A from Bitcoin StackExchange
+
+*[Bitcoin StackExchange][bitcoin.se] is one of the first places Optech
+contributors look for answers to their questions---or when we have a
+few spare moments of time to help curious or confused users.  In
+this monthly feature, we highlight some of the top-voted questions and
+answers posted since our last update.*
+
+{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{%
+endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [Would a schnorr pubkey be a different length than a taproot pubkey like P2WPKH and P2WSH?]({{bse}}91531)
+  Murch explains that unlike segwit v0 which has different P2WPKH and P2WSH
+  output types and lengths, all segwit v1 Pay-to-Taproot (P2TR) outputs are
+  always the same length.
+
+- [MuSig Signature Interactivity]({{bse}}91534)
+  Justinmoon asks why MuSig signing is always interactive and about secure,
+  offline interactive signing. Nickler explains each of the rounds involved
+  with MuSig signing as well as some pitfalls to avoid during signing.
+
+- [How does the bech32 length-extension mutation weakness work?]({{bse}}91602)
+  Jnewbery asks about a bech32 weakness where adding or removing q characters
+  immediately before the final p character of an address ending with p can make
+  a new, valid bech32 address, eluding the some of the bech32 error detection
+  properties. Pieter Wuille provides some algebraic details and MCCCS explains
```

Comment:
> elusing some of the bech32 error detection properties

I'm not sure I'd phrase it this way. The error detection properties of bech32 work as designed _under the assumption that the length is fixed_. I'd either remove this entirely or reword it to something about "people's expectations of the error detection properties" or similar.

Reply:
@jnewbery Thanks for that feedback. I think the changes made by @harding (thanks for those enhancements!) in 8f6841e9586692f1b85cd4e54b1f99a11355561a look good to me.